docs/noise.md

```diff
@@ -0,0 +1,121 @@
+Currently Cirq supports modeling noise via *operator sum* representations of
+noise (these evolutions are also known as quantum operations or quantum
+dynamical maps).  This formalism models evolution of the density matrix
+via
+
+![Operator sum representation: $\rho \rightarrow \sum_k A_k \rho A_k^\dagger$](resources/OperatorSumDef.gif)
+
+Where here the the A<sub>k</sub> are *Krauss* operators. These operators
+are not necessarily unitary and must satisfy the trace preserving
+property
+
+![Operator sum normalization: $\sum_k A_k^\dagger A_k = I$](resources/OperatorSumNormDef.gif)
+
+### Magic methods
+
+A `Gate` can represent a operator sum representation by supporting the
+`channel` protocol.  Alternatively, for channels that represent probabilistic
+mixtures of unitaries, one can implement the `mixture` protocol.
+
+#### cirq.channel and def _channel_
+
+To represent an operator sum evolution a `Gate` should implement the
+`SupportsChannel` protocol.  To do this the `Gate` should implement the
+`_channel_(self) -> Sequence[np.ndarray]:` method.  This method should return
+the sequence of `numpy` matrices corresponding to the Krauss operators. The
+basis in which this matrix is expressed is always implicit with respect to
+the object being called. For example, for GateOperations these matrices
+must be ordered with respect to the list of qubits that the channel is
+applied to. The qubit-to-amplitude order mapping matches the
+ordering of `numpy.kron(A, B)`, where `A` is a qubit earlier in the list
+than the qubit `B`.
+
+In one has defined `_channel_`, then that `Gate` and any `GateOperation`
+that uses that gate can be used as an argument to `cirq.channel` and
+`cirq.channel` will return this sequence of matrices.
+
+Besides objects that support `_channel_`, `cirq.channel` will also fall
+back to other objects that can be interpreted as channels.  For example if a
+channel is a probabilistic mixture of unitary gates (see below), then
+`cirq.channel` will fall back to seeing of the object supports `_mixture_`.
+If `_mixture_` is not supported, then `cirq.channel` checks to see if
+`_unitary_` is supported.
+
+In addition to supporting `_channel_`, objects that are channels should also
+implement `_has_channel_(self) -> bool` to return True.  This method is
+used to quickly determine whether an object has a `_channel_` or not without
+having to do the potentially expensive creation of the matrices for the
+channel.
+
+#### cirq.mixture, cirq.mixture_channel, and def _mixture_
+
+Some channels can be interpreted as probabilistically selecting between
+different unitary evolutions.
+
+![Mixture channel: $\rho \rightarrow \sum_k p_k U_k \rho U_k^\dagger {\rm ~where~} \sum_k p_k =1 {\rm ~and~ U_k U_k^\dagger= I}$](resources/MixtureChannelDef.gif)
+
+In this case it is possible to perform
+monte carlo simulations of these gates using a wave function based simulator
+(and not a density matrix based simulator).  In this case instead of
```

Comment:
Maybe "and not a density matrix based simulator" -> "and not a slower density matrix based simulator"? (add "slower" to explain why one would want this).